saleor/checkout/complete_checkout.py

```diff
@@ -1291,6 +1296,11 @@ def complete_checkout(
             order = Order.objects.get_by_checkout_token(checkout_info.checkout.token)
             return order, False, {}
 
+        # We need to refetch the checkout lines to ensure that we process checkout
+        # for correct data.
+        # Lines are selected for update by checkout select for update in line 1421
+        _refetch_checkout_lines(checkout, checkout_info, discounts, manager)
```

Comment:
Aren't we reintroducing a slow 3rd party call inside a transaction? From what I see `_refetch_checkout_lines` involves synchronous webhooks which do not belong inside DB transactions

Reply:
Hm, so maybe it should be refetched just before the transaction block. What do you think?

Reply:
Yes, it will be a little chance that something will change. We cannot add the lock for the entire checkout process as the stocks will be frozen too.